src/main/java/com/gmail/goosius/siegewar/command/SiegeWarAdminCommand.java

```diff
@@ -244,51 +244,74 @@ private void parseSiegeWarSiegeImmunityCommand(CommandSender sender, String[] ar
-			Messaging.sendMsg(sender, Translation.of("msg_set_siege_immunities_town", args[1], args[2]));
-
-		} else if (args.length >= 3 && args[0].equalsIgnoreCase("nation")) {
+			if (args[2].equalsIgnoreCase("permanent")) {
+				TownMetaDataController.setSiegeImmunityEndTime(town, -1l);
+				timeDuration = "permanent";
+			} else {
+				long durationMillis = (long)(Long.parseLong(args[2]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
+				TownMetaDataController.setSiegeImmunityEndTime(town, System.currentTimeMillis() + durationMillis);
+				timeDuration = (Long.parseLong(args[2]) == 1L) ? "1 hour" : String.format("%s hours", args[2]);
+			}
+			TownyMessaging.sendPrefixedTownMessage(town, Translation.of("msg_set_siege_immunities_town", args[1], timeDuration));
+			Messaging.sendMsg(sender, Translation.of("msg_set_siege_immunities_town", args[1], timeDuration));
+		}
+		else if (args.length >= 3 && args[0].equalsIgnoreCase("nation")) {
 			//nation {nationname} {hours}
 			Nation nation = TownyUniverse.getInstance().getNation(args[1]);
 			if (nation == null) {
 				Messaging.sendErrorMsg(sender, Translation.of("msg_err_not_registered_1", args[1]));
 				return;
 			}
-			long durationMillis = (long)(Long.parseLong(args[2]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
+			long endTime;
+			if (args[2].equalsIgnoreCase("permanent")) {
+				endTime = -1l;
+				timeDuration = "permanent";
+			} else {
+				long durationMillis = (long)(Long.parseLong(args[2]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
+				endTime = System.currentTimeMillis() + durationMillis;
+				timeDuration = (Long.parseLong(args[2]) == 1L) ? "1 hour" : String.format("%s hours", args[2]);
+			}
 			for (Town town : nation.getTowns()) {
-				TownMetaDataController.setSiegeImmunityEndTime(town, System.currentTimeMillis() + durationMillis);
+				TownMetaDataController.setSiegeImmunityEndTime(town, endTime);
 			}
-			TownyMessaging.sendPrefixedNationMessage(nation, Translation.of("msg_set_siege_immunities_nation", args[1], args[2]));
-			Messaging.sendMsg(sender, Translation.of("msg_set_siege_immunities_nation", args[1], args[2]));
+			TownyMessaging.sendPrefixedNationMessage(nation, Translation.of("msg_set_siege_immunities_nation", args[1], timeDuration));
+			Messaging.sendMsg(sender, Translation.of("msg_set_siege_immunities_nation", args[1], timeDuration));
 
 		} else if(args.length >= 2 && args[0].equalsIgnoreCase("alltowns")) {
 			//all towns
-			long durationMillis = (long)(Long.parseLong(args[1]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
+			long endTime;
+			if (args[1].equalsIgnoreCase("permanent")) {
+				endTime = -1l;
+				timeDuration = "permanent";
+			} else {
+				long durationMillis = (long)(Long.parseLong(args[1]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
+				endTime = System.currentTimeMillis() + durationMillis;
+				timeDuration = (Long.parseLong(args[1]) == 1L) ? "1 hour" : String.format("%s hours", args[1]);

```

Comment:
:construction: 
- This formula needs a fix, because it has hardcoded english strings 
- Also it doesn't need to account for single hours (*but you can support this is you wish*)
- I believe something like this would work: 
>   timeDuration = Long.parseLong(args[1]) + " " + Towny.Translation.of("Hours").